BootloaderCorePkg/Stage1A/Ia32/SecEntry.nasm

```diff
@@ -95,14 +95,20 @@ CheckStackRangeDone:
         bts     ebx, 0               ; Set BIT0 in Status
 
 CheckStatusDone:
+        ;
         ; Setup HOB
+        ; This structure has to match the layout of STAGE1A_ASM_PARAM
+        ;
+        push    $0
         push    ebx                  ; Status
         push    edi                  ; TimeStamp[0] [63:32]
         push    esi                  ; TimeStamp[0] [31:0]
         push    edx                  ; CarTop
         push    ecx                  ; CarBase
+        push    $0                   ; Keep the stack 16-byte aligned
```

Comment:
In your commit msg, you mentioned "Version 1.0 of the System V Intel386 ABI supplement" requirement.  Do you know it has the 16 byte requirement for x86 ?

Reply:
A copy of this ABI supplement is here:

        https://www.uclibc.org/docs/psABI-i386.pdf

Which states:

```
2.2.2 The Stack Frame

In addition to registers, each function has a frame on the run-time
stack. This stack grows downwards from high addresses. Table 2.2 shows
the stack organization.  The end of the input argument area shall be
aligned on a 16 (32, if __m256 is passed on stack) byte boundary.
```

You could argue that Slimbootloader doesn't need to follow this ABI,
which is a fair point -- it's a custom OS of sorts, after all.  But
in that case, it probably should not just be using the host system's
C compiler, either, because that compiler might expect that this ABI
is followed.

In my particular case, I was trying to build Slimbootloader on a Linux
system where the distribution-provided `gcc` enables SSE by default,
which it is allowed to do, but which doesn't work with Slimbootloader,
because Slimbootloader doesn't provide 16-byte stack alignment, and
using this compiler I get random #GP faults in Slimbootloader on SSE
instructions generated by the compiler that have unaligned memory
arguments on the stack.  In other words, Slimbootloader is using a
compiler to build itself that requires a certain ABI without providing
that ABI.  (Hence my previous patch to dump out the instruction bytes
on a fatal exception. :) )

The way other projects solve this is providing their own toolchain,
which is built with the settings and the ABI that the project expects.
For example, the Zephyr real-time operating system ships with its own 
toolchains.  

I figured that satisfying the ABI guarantees likely expected by the
system compiler is probably the easier option in this case -- it only
requires this change plus a one-line change to
MdePkg/Library/BaseLib/Ia32/InternalSwitchStack.nasm to guarantee
16-byte stack alignment for Stage1B/2, for which I wanted to submit a
patch as well.

Reply:
👍  That makes sense to me.   Thank you for the detailed explanation.  
I am not aware of this issue.  SBL build infrastructure is fully aligned with EDK2.  I am not sure if EDK2 has applied the similar for stack pointer init.  But I agree with you that SBL should provide better compatibility for different toolchain.